### PR TITLE
Polish deploy-multi-chain script

### PR DESCRIPTION
**Pull request by smol-ninja**
### There is one major issue with deployment:
1. `SablierV2LockupDynamic` has a contract size of **25.224 kB**
2. `SablierV2NFTDescriptor` has a contract size of **25.661 kB**

Since both of them are above 24576 bytes, they fail to deploy. Failed [Tx #1](https://sepolia.etherscan.io/tx/0x8186c7b157a93617ce060b78dc7e2ac9efc55f9ff384925e41e580f877ee4621) and [Tx #2](https://sepolia.etherscan.io/tx/0xa09f96ad683c108c1cf02a1125fdd9c540f6cf3a053ddfbbadd401bc630b8320)

Changes with this PR
-------
### .evn.example
- Rename "BINANCE_RPC_URL" to "BSC_RPC_URL"
- Add "SEPOLIA_RPC_URL" and "SEPOLIA_ADMIN"

### shell/deploy-multi-chains.sh
- Remove redundant "GOERLI_CHAIN_ID"
- Load variables from ".env" file
- Add `ON_ALL_CHAINS` bool to avoid duplication of chains in `requested_chains` associative array in case of accidental chain names args along with `--all` flag
- Replace `ChainID $chain_id, Version 1.1.0` with `ChainID_${chain_id}_Version_1.1.0`. Seems like https://github.com/foundry-rs/foundry/issues/6265 still exists. In the latest build, Foundry fails to parse**empty spaces (" ")** and **comma (",")**
- Added color codes for friendly output

Screenshot 2023-12-12 at 16 34 35
Screenshot 2023-12-12 at 16 18 52

### CODE REVIEW

**Comment by smol-ninja** [comment]
CI failed. It couldn't find "node_modules/forge-std/src/Script.sol" even though `build` successfully completed with `pnpm install`. Any idea?

**Comment by andreivladbrg** [comment]
> There is one major issue with deployment:
    SablierV2LockupDynamic has a contract size of 25.224 kB
    SablierV2NFTDescriptor has a contract size of 25.661 kB

I am going to review this tomorrow, but I want to clarify this now. This issue appeared due to different `optimizer_runs` config in the `foundry.toml` file:

https://github.com/sablier-labs/v2-periphery/blob/6a3ca3171802677517005831fc85b618a655acf1/foundry.toml#L13

https://github.com/sablier-labs/v2-core/blob/63204c9ab44e6f5efcb0c38249f229f7d4affe73/foundry.toml#L14

Forgot about this.

This means we can't run a deploy protocol script. We will have to run the [deploy core](https://github.com/sablier-labs/v2-core/blob/build/multi-chain-script/shell/deploy-multi-chains.sh) script there.

**Comment by smol-ninja** [comment]
I see. In that case, I have updated the script to deploy SablierV2Batch and SablierV2MerkleStream: https://github.com/sablier-labs/v2-periphery/pull/229/commits/a0e8b39e150395290619474d45f8ebb76a11c3de

Also, I think we should include `--verifier-url` flag as well. The verifier url is not same for all networks: https://docs.etherscan.io/getting-started/endpoint-urls. Without that, the script may not necessarily be able to verify contracts on etherscan.

**Comment by andreivladbrg** [comment]
Thanks for the PR

> Foundry fails to parseempty spaces (" ") and comma (",")

regarding this, a fix would be using "`' '`": `'ChainID $chain_id, Version 1.1.0'` 

>Added color codes for friendly output

it definitely looks better now, love it.

since we are not going to use the deploy protocol anymore, we can remove the admin env variables

> CI failed. It couldn't find "node_modules/forge-std/src/Script.sol" even though build successfully completed with pnpm install. Any idea?

this is because the `pnpm install` step was not added in the `ci` for all jobs, fixed this: https://github.com/sablier-labs/v2-periphery/commit/36a58a894245d9632f28c4bdc0cc2c4b292d8fb8

> think we should include --verifier-url flag as well. The verifier url is not same for all networks: https://docs.etherscan.io/getting-started/endpoint-urls.

this flag is used for [custom networks](https://book.getfoundry.sh/reference/cli/forge/verify-contract?highlight=--verifier-url#forge-verify-contract). when running the script and you pass a rpc url, it automatically knows its verifier, thus you only need the api key

**Comment by andreivladbrg** [comment]
pushed a new commit, lmk if it looks good

**Comment by smol-ninja** [comment]
LGTM!

File: shell/deploy-multi-chains.sh [hunk 1]
```diff
@@ -122,50 +133,46 @@ FOUNDRY_PROFILE=optimized forge build
...
         # Construct the command
-        deployment_command="forge script script/DeployDeterministicProtocol2.s.sol \
+        deployment_command="forge script script/DeployDeterministicPeriphery.s.sol \
         --rpc-url $rpc_url \
-        --sig run(string,address,address) \
-        \"ChainID $chain_id, Version 1.1.0\" \
-        $admin \
-        $comptroller \
+        --sig run(string) \
+        \"ChainID_${chain_id}_Version_1.1.0\" \
```
Comment (smol-ninja) [inline shell/deploy-multi-chains.sh:153]: Rename to 'ChainID ${chain_id}, Version 1.1.0'

**Comment by andreivladbrg** [comment]
> LGTM!

great, would mind if you would update the [core script](https://github.com/sablier-labs/v2-core/tree/build/multi-chain-script) as well with these:
>>Load variables from ".env" file
Add ON_ALL_CHAINS bool to avoid duplication of chains in requested_chains associative array in case of accidental chain names args along with --all flag
Replace ChainID $chain_id, Version 1.1.0 with ChainID_${chain_id}_Version_1.1.0.
Added color codes for friendly output

**Review by andreivladbrg — APPROVED** [review]

**Comment by PaulRBerg** [comment]
> Seems like https://github.com/foundry-rs/foundry/issues/6265 still exists. In the latest build, Foundry fails to parseempty spaces (" ") and comma (",")

In future scenarios like this, @smol-ninja, can you please the external GitHub issue?

The issue should have been re-opened - and we should have told the Foundry maintainers about this.

**Comment by PaulRBerg** [comment]
> This means we can't run a deploy protocol script. We will have to run the [deploy core](https://github.com/sablier-labs/v2-core/blob/build/multi-chain-script/shell/deploy-multi-chains.sh) script there and here on periphery we should run deploy periphery.

We can run it now since the `test/utils` are part of the NPM package. References:

- https://github.com/sablier-labs/v2-periphery/pull/91#issuecomment-1580391265
- https://github.com/sablier-labs/v2-core/discussions/662

**Comment by smol-ninja** [comment]
> In future scenarios like this, @smol-ninja, can you please also update the external GitHub issue?

Yeah sure. Not sure why I didn't do that but I acknowledge that I should have mentioned it there too.

**Comment by PaulRBerg** [comment]
No worries, thanks Shub.

**Pull request merged**